Lib/textwrap.py

```diff
@@ -426,40 +427,10 @@ def dedent(text):
     """
     # Look for the longest leading string of spaces and tabs common to
     # all lines.
-    margin = None
-    text = _whitespace_only_re.sub('', text)
-    indents = _leading_whitespace_re.findall(text)
-    for indent in indents:
-        if margin is None:
-            margin = indent
-
-        # Current line more deeply indented than previous winner:
-        # no change (previous winner is still on top).
-        elif indent.startswith(margin):
-            pass
-
-        # Current line consistent with and no deeper than previous winner:
-        # it's the new winner.
-        elif margin.startswith(indent):
-            margin = indent
-
-        # Find the largest common whitespace between current line and previous
-        # winner.
-        else:
-            for i, (x, y) in enumerate(zip(margin, indent)):
-                if x != y:
-                    margin = margin[:i]
-                    break
-
-    # sanity check (testing/debugging only)
-    if 0 and margin:
-        for line in text.split("\n"):
-            assert not line or line.startswith(margin), \
-                   "line = %r, margin = %r" % (line, margin)
 
-    if margin:
-        text = re.sub(r'(?m)^' + margin, '', text)
-    return text
+    warnings.warn("textwrap.dedent is deprecated, use the str.dedent method instead",
+            PendingDeprecationWarning)
```

Comment:
Please use `stacklevel=2` to make sure the relevant call site from user program is printed instead of this line in the traceback. Please align `PendingDeprecationWarning` with the brace.